libnd4j/include/ops/declarable/helpers/cpu/scatter.cpp

```diff
@@ -27,6 +27,49 @@ namespace nd4j    {
 namespace ops     {
 namespace helpers {
 
+///////////////////////////////////////////////////////////////////
+// x - indices, z - input/output
+template<typename T>
+Nd4jLong checkIndices_(const NDArray& indices, const NDArray& output, const int axis) {
+
+    std::atomic<int64_t> numOfBadIndx{0};
+
+    const auto x = indices.bufferAsT<T>();
+
+    const auto xShapeInfo = indices.getShapeInfo();
+    const auto zShapeInfo = output.getShapeInfo();
+
+    const auto xRank = indices.rankOf();
+
+    auto func = PRAGMA_THREADS_FOR {
+
+        Nd4jLong* xCoords = new Nd4jLong[xRank];
```

Comment:
Just make it Nd4jLong[MAX_RANK] like we usually do, so it'll be on-heap allocation.